### PR TITLE
feat: 'StorageValuePipeline' should process large-size content without buffering it all in memory

### PR DESCRIPTION
**Pull request by dgarus**
Closes: #432 

- implemented `StorageValuePipeline#processWithResult` using RxJava
- added test storage that can emit chunks of data with random size and random delays
- added `UncheckedRunnable` and static build method for running with `IOException`

### CODE REVIEW

**Comment by dgarus** [comment]
@olenagerasimova @g4s8 hello!
Could you please review the PR?

**Review by olenagerasimova — APPROVED** [review]
@dgarus thanks, LGTM and I'd suggest to remove rio dependency as we don't use it any more

**Comment by dgarus** [comment]
@olenagerasimova thank you for review!
> I'd suggest to remove rio dependency

At least `FileStorage` uses rio.

**Comment by olenagerasimova** [comment]
@dgarus oh, ok then. Can I merge or wait for another review?

**Comment by dgarus** [comment]
@olenagerasimova, let's wait @g4s8 review.

**Review by g4s8 — COMMENTED** [review]
@dgarus it's quite hard to review custom implementations of publishers/subscribers by human, this is why reactivestreams has [TCK](https://www.reactive-streams.org/reactive-streams-tck-1.0.4-javadoc/org/reactivestreams/tck/package-summary.html) which checks that new reactive primitive implements [reactive specification](https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.4/README.md#specification) correctly. I'd highly recommend to run `SubscriberWhiteboxVerification` test with new `DefaultSubscriber` implementation. For reference you can check my example here: https://github.com/cqfn/rio/blob/master/src/test/java/org/cqfn/rio/channel/WritableChannelSubscriberTest.java

**Comment by g4s8** [comment]
@dgarus thanks, looks good. Also, I'd add Testng runner for Maven surefire configuration, because it doesn't run Junit and Testng suites simultaneously.

**Comment by dgarus** [comment]
@g4s8 thank you!
To run test by maven, it was sufficient to rename test
`ContentAsInputStreamWhiteboxVerification` -> `ContentAsInputStreamWhiteboxVerificationTest`.

**Review by olenagerasimova — APPROVED** [review]
@dgarus can I merge?

**Comment by dgarus** [comment]
> @dgarus can I merge?

yes, thank you!

**Pull request merged**
